src/components/create-study-form.js

```diff
@@ -222,21 +220,43 @@ export const CreateStudyForm = ({ open, onClose }) => {
         const name = e.target.value;
         setStudyName(name);
 
-        setStudyNameChecked(false);
         setLoadingCheckStudyName(true);
```

Comment:
Maybe we should move this call at the beginning of `updateStudyFormState` ?

Reply:
No it should not, if we do so (as the checks are really fast) we can't even see the loading sign. The point of it is to show that something is happening as soon as it starts to load